src/components/actions/executeAction.ts

```diff
@@ -17,10 +23,11 @@ const graphQLEndpoint = config.graphQLendpoint
 export async function executeAction(
   payload: ActionPayload,
   actionLibrary: ActionLibrary,
-  additionalObjects: any = {}
+  additionalObjects: any = {},
+  previewData?: Partial<ActionApplicationData>
 ): Promise<ActionQueueExecutePayload> {
-  // Get fresh applicationData for each Action
-  const applicationData = await getApplicationData({ payload })
+  // Get fresh applicationData for each Action, and inject previewData if present
+  const applicationData = merge(await getApplicationData({ payload }), previewData)
```

Comment:
Need to use lodash merge not just spread operators, cause we will sometimes need to override deep properties without obliterating the other properties in the same deep object -- spread operators only go one level deep.

Reply:
I think this is kind of dangerous, would expect previewData to override the whole object, otherwise it's hard to think about permutations of combination of object (and no way to actually override the whole object if needed). What would be a use case for overriding something that's deep nested ?

Reply:
In applicationData.reviewData.reviewDecision -- we want to set reviewDecision.decision without destroying the whole "reviewData" object within applicationData. This is the exact case I came across that prompted me to make this change.

Reply:
You can still override the whole object if you set "applicationData.reviewData" in override props, but this allows us to surgically target a specific deeply-nested property if we need to without destroying its sibling properties (but we still can if we want to as well)

Reply:
Also, in "generateDoc", for the LOQ document, we need to pass in a few extra "additionalData" props in without removing the existing ones.

It's demonstrated in the front-end of this PR, with the associated snapshot.

Just out of interest, why would you ever NOT want to do this? In which use case would you want to pass in a deeply nested property path and have it destroy all the parent property objects on the way down?

Reply:
> Just out of interest, why would you ever NOT want to do this? In which use case would you want to pass in a deeply nested property path and have it destroy all the parent property objects on the way down?

This makes sense (you are right for the previewData override, it makes sense to override in a nested way)
I think I was more thinking about the other example with parameter_queries override, like what happens when evaluator expressions get merged